circleci: failed tests (13046): TestClusterRecovery

The following test appears to have failed:

[#13046](https://circleci.com/gh/cockroachdb/cockroach/13046):

```
I0219 16:55:12.282009 28384 acceptance/chaos_test.go:170  client 0 shutting down
I0219 16:55:12.287312 28384 acceptance/chaos_test.go:170  client 1 shutting down
I0219 16:55:12.287367 28384 acceptance/chaos_test.go:170  client 2 shutting down
W0219 16:55:12.337542 28384 acceptance/chaos_test.go:242  round 2: cluster recovered
I0219 16:55:12.437925 28384 acceptance/cluster/localcluster.go:583  stopping
--- FAIL: TestClusterRecovery (16.47s)
	chaos_test.go:147: dial tcp 0.0.0.0:32799: getsockopt: connection refused
=== RUN   TestNodeRestart
I0219 16:55:14.733030 28384 acceptance/cluster/localcluster.go:259  started docker-spy: 172.17.0.25
I0219 16:55:14.733056 28384 acceptance/cluster/localcluster.go:266  initializing cluster
I0219 16:55:16.353571 28384 acceptance/cluster/localcluster.go:519  creating certs (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.930358903
I0219 16:55:17.916638 28384 acceptance/cluster/localcluster.go:437  *** started roach0.local ***
  ui:    https://0.0.0.0:32801
  trace: https://0.0.0.0:32801/debug/requests
  logs:  /tmp/circle-artifacts.6eCohat/acceptance/TestNodeRestart/roach0.local/cockroach.INFO
  pprof: docker exec -it 0c91b /bin/bash -c 'go tool pprof /cockroach <(wget --no-check-certificate -qO- https://$(hostname):26257/debug/pprof/heap)'
--
--- SKIP: TestBuildBabyCluster (0.00s)
	terraform_test.go:32: only enabled during testing
=== RUN   TestFiveNodesAndWriters
--- SKIP: TestFiveNodesAndWriters (0.00s)
	main_test.go:63: running in docker mode
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

--- a/acceptance/chaos_test.go
+++ b/acceptance/chaos_test.go
@@ -31,6 +31,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/acceptance/cluster"
 	"github.com/cockroachdb/cockroach/testutils"
+	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/log"
 	"github.com/cockroachdb/cockroach/util/randutil"
 	"github.com/cockroachdb/cockroach/util/stop"
@@ -134,19 +135,28 @@ func transferMoney(client *testClient, numAccounts, maxTransfer int) error {
 	return err
 }
 
+func retryableError(err error) bool {
+	return testutils.IsError(err, "connection reset by peer") || testutils.IsError(err, "connection refused") || testutils.IsError(err, "failed to send RPC") || testutils.IsError(err, "EOF")
+}
+
 // Verify accounts.
 func verifyAccounts(t *testing.T, client *testClient) {
-	// Hold the read lock on the client to prevent it being restarted by
-	// chaos monkey.
 	var sum int
-	client.RLock()
-	defer client.RUnlock()
-	if err := client.db.QueryRow("SELECT SUM(balance) FROM bank.accounts").Scan(&sum); err != nil {
-		t.Fatal(err)
-	}
+	util.SucceedsSoon(t, func() error {
+		// Hold the read lock on the client to prevent it being restarted by
+		// chaos monkey.
+		client.RLock()
+		defer client.RUnlock()
+		err := client.db.QueryRow("SELECT SUM(balance) FROM bank.accounts").Scan(&sum)
+		if err != nil && !retryableError(err) {
+			t.Fatal(err)
+		}
+		return err
+	})
 	if sum != 0 {
 		t.Fatalf("The bank is not in good order. Total value: %d", sum)
 	}
+
 }
 
 // Continuously transfers money until done().
@@ -155,7 +165,7 @@ func transferMoneyLoop(idx int, state *testState, numAccounts, maxTransfer int) 
 	for !state.done() {
 		if err := transferMoney(client, numAccounts, maxTransfer); err != nil {
 			// Ignore some errors.
-			if !testutils.IsError(err, "connection refused") && !testutils.IsError(err, "failed to send RPC") {
+			if !retryableError(err) {
 				// Report the err and terminate.
 				state.errChan <- err
 				break
@@ -297,7 +307,7 @@ func waitClientsStop(num int, state *testState, stallDuration time.Duration) {
 // being killed and restarted continuously. The test doesn't measure write
 // performance, but cluster recovery.
 func TestClusterRecovery(t *testing.T) {
-	t.Skipf("TODO(pmattis): #4517")
+	runTestOnConfigs(t, testClusterRecoveryInner)
 }
 
 func testClusterRecoveryInner(t *testing.T, c cluster.Cluster, cfg cluster.TestConfig) {
@@ -357,7 +367,6 @@ func testClusterRecoveryInner(t *testing.T, c cluster.Cluster, cfg cluster.TestC
 // than the one the client is connected to is being restarted periodically.
 // The test measures read/write performance in the presence of restarts.
 func TestNodeRestart(t *testing.T) {
-	t.Skipf("TODO(pmattis): #4517")
 	runTestOnConfigs(t, testNodeRestartInner)
 }
 

